practice_points.c

```diff
@@ -0,0 +1,107 @@
+#include <cstring>
+#include <memory>
+
+typedef struct point
+{
+    int x, y;
+} POINT;
+
+// a segment is a string like: (x, y)
+// eg: (10, 20)
+// eg: (5123, -15123)
+POINT segment_to_point(char* segment)
+{
+    // buffer will contain the segment without the parentheses:
+    int buffer_length = strlen(segment) - 2;
+    char* buffer = (char*)malloc(sizeof(char) * (buffer_length + 1));
+    memcpy(buffer, segment + 1, buffer_length);
+    buffer[buffer_length] = '\0';
+
+    // find the comma which separates x and y:
+    // (we could also use strtok, but since there's just two parts, this is more simple)
+    for (int i = 0; i != buffer_length; ++i)
+    {
+        if (buffer[i] == ',')
+        {
+            POINT p;
+            p.x = atoi(buffer);         // reads digits until the comma (automatically stops)
+            p.y = atoi(buffer + i + 1); // reads digits after the comma until the end
+            return p;
+        }
+    }
+
+    // in case there is no comma:
+    // (this should never happen)
+    return POINT();
```

Comment:
`POINT()` isn't a function. C compiler refuses it.
```suggestion
    POINT p = { 0, 0 };
    return p;
```

Reply:
I'm not super sure about this one. How do you want to handle the error?